src/Symfony/Component/VarDumper/VarDumper.php

```diff
@@ -27,34 +34,87 @@
 class VarDumper
 {
     private static $handler;
+    private static $locked = false;
 
-    public static function dump($var)
+    /**
+     * @final
+     *
+     * @param bool $lock lock the registered handler so it cannot be replaced
+     */
+    public static function register(bool $lock = false): void
```

Comment:
```suggestion
    final public static function register(bool $lock = false): void
```

Reply:
I marked it `@final` though. I don't remember the reasons for using `@final` over enforcing it for a new method, if there are any?

Reply:
Usually we do that on existing methods, to not break the BC and to provide a migration path.
On new class we can do that directly I guess.
(Or may be to be able to mock method...)